pkg/apis/rook.io/v1/types.go

```diff
@@ -221,6 +221,10 @@ type VolumeSource struct {
 	// +kubebuilder:validation:Pattern=`^([0-9]*[.])?[0-9]$`
 	// +optional
 	CrushInitialWeight string `json:"crushInitialWeight,omitempty"`
+	// CrushPrimaryAffinity represents OSD primary-affinity within range [0,1)
```

Comment:
question: does a `(` closing parentheses signify something different here? 

```suggestion
	// CrushPrimaryAffinity represents OSD primary-affinity within range [0,1]
```

Reply:
@sp98 The default value of primary-affinity is 1. Thus, if setting explicit value, it should be within the range [0, 1) -- that is, excluding 1.

Reply:
@sp98 Please note also that this affects the kubebuilder validation pattern